scenario/mocking.py

```diff
@@ -727,21 +741,39 @@ def _layers(self) -> Dict[str, pebble.Layer]:
 
     @property
     def _service_status(self) -> Dict[str, pebble.ServiceStatus]:
-        return self._container.service_status
+        return self._container.service_statuses
+
+    # Based on a method of the same name from ops.testing.
+    def _find_exec_handler(self, command) -> Optional["Exec"]:
+        handlers = {exec.command_prefix: exec for exec in self._container.execs}
+        # Start with the full command and, each loop iteration, drop the last
+        # element, until it matches one of the command prefixes in the execs.
+        # This includes matching against the empty list, which will match any
+        # command, if there is not a more specific match.
+        for prefix_len in reversed(range(len(command) + 1)):
+            command_prefix = tuple(command[:prefix_len])
+            if command_prefix in handlers:
+                return handlers[command_prefix]
+        # None of the command prefixes in the execs matched the command, no
+        # matter how much of it was used, so we have failed to find a handler.
+        return None
 
-    def exec(self, *args, **kwargs):  # noqa: U100 type: ignore
-        cmd = tuple(args[0])
-        out = self._container.exec_mock.get(cmd)
-        if not out:
+    def exec(self, command, **kwargs):  # noqa: U100 type: ignore
+        handler = self._find_exec_handler(command)
+        if not handler:
             raise RuntimeError(
-                f"mock for cmd {cmd} not found. Please pass to the Container "
-                f"{self._container.name} a scenario.ExecOutput mock for the "
+                f"mock for cmd {command} not found. Please pass to the Container "
+                f"{self._container.name} a scenario.Exec mock for the "
```

Comment:
we could add here a copy-pastable snippet of what the input could look like?

`Container(..., execs=\{scenario.Exec(command_prefix={list(command)}), \})`